.github/workflows/golangci-lint.yml

```diff
@@ -4,6 +4,8 @@ on:
     branches:
       - main
       - master
+  pull_request_target:
+    types: [ opened, edited, reopened ]
```

Comment:
I might have to double check this later, but iirc we should opt for pull_request or pull_request_target. The latter is a bit more secure when it comes to secrets. Although I want to make sure we can also pick branches.

Reply:
Yes, we can use either `pull_request` or `pull_request_target`.
`pull_request_target` is more secure because that trigger doesn't have write access to the code and also read access to secrets.

Reply:
> In order to solve this, we’ve added a new pull_request_target event, which behaves in an almost identical way to the pull_request event with the same set of filters and payload. However, instead of running against the workflow and code from the merge commit, the event runs against the workflow and code from the base of the pull request. This means the workflow is running from a trusted source and is given access to a read/write token as well as secrets enabling the maintainer to safely comment on or label a pull request. This event can be used in combination with the private repository settings as well.

https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/

Reply:
Because this workflow doesn't perform either write operations or accesses any secret, either `pull_request` or `pull_request_target` may be used.

Reply:
I don't think we need this change, I opened a new PR #4381 and the lint action is running. I think we just need to have all existing PRs rebase and/or re-open the PRs.

/close